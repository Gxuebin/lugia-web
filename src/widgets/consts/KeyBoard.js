/**
 * Created by liguoxin on 16/2/5.
 */

const Keys = {
  BACKSPACE: 8,

  TAB: 9,

  NUM_CENTER: 12,

  ENTER: 13,

  RETURN: 13,

  SHIFT: 16,

  CTRL: 17,
  CONTROL: 17,

  ALT: 18,

  PAUSE: 19,

  CAPS_LOCK: 20,

  ESC: 27,

  SPACE: 32,

  PAGEUP: 33,

  PAGEDOWN: 34,

  END: 35,

  HOME: 36,

  LEFT: 37,

  UP: 38,

  RIGHT: 39,

  DOWN: 40,

  PRINT_SCREEN: 44,

  INSERT: 45,

  DELETE: 46,

  ZERO: 48,

  ONE: 49,

  TWO: 50,

  THREE: 51,

  FOUR: 52,

  FIVE: 53,

  SIX: 54,

  SEVEN: 55,

  EIGHT: 56,

  NINE: 57,

  A: 65,

  B: 66,

  C: 67,

  D: 68,

  E: 69,

  F: 70,

  G: 71,

  H: 72,

  I: 73,

  J: 74,

  K: 75,

  L: 76,

  M: 77,

  N: 78,

  O: 79,

  P: 80,

  Q: 81,

  R: 82,

  S: 83,

  T: 84,

  U: 85,

  V: 86,

  W: 87,

  X: 88,

  Y: 89,

  Z: 90,

  CONTEXT_MENU: 93,

  NUM_ZERO: 96,

  NUM_ONE: 97,

  NUM_TWO: 98,

  NUM_THREE: 99,

  NUM_FOUR: 100,

  NUM_FIVE: 101,

  NUM_SIX: 102,

  NUM_SEVEN: 103,

  NUM_EIGHT: 104,

  NUM_NINE: 105,

  NUM_MULTIPLY: 106,

  NUM_PLUS: 107,

  NUM_MINUS: 109,

  NUM_PERIOD: 110,

  NUM_DIVISION: 111,

  F1: 112,

  F2: 113,

  F3: 114,

  F4: 115,

  F5: 116,

  F6: 117,

  F7: 118,

  F8: 119,

  F9: 120,

  F10: 121,

  F11: 122,

  F12: 123,
  FengHao: 186,
  JiaHao: 187,
  DouHao: 188,
  JianHao: 189,
  JuHao: 190,
  XieGang: 191,
  BLH: 192,
  ZKH: 219,
  SX: 220,
  YKH: 221,
  DYH: 222,
};

const safariKeys = {
  3: 13,
  63234: 37,
  63235: 39,
  63232: 38,
  63233: 40,
  63276: 33,
  63277: 34,
  63272: 46,
  63273: 36,
  63275: 35,
};

export function isNavKeyPress(e) {
  const k = normalizeKey(e.keyCode);
  return (k >= 33 && k <= 40) || k == Keys.RETURN || k == Keys.TAB || k == Keys.ESC;
}

export function isSpecialKey(e) {
  const k = normalizeKey(e.keyCode);
  return (
    (e.type == 'keypress' && e.ctrlKey) ||
    isNavKeyPress(e) ||
    k == Keys.BACKSPACE ||
    (k >= 16 && k <= 20) ||
    (k >= 44 && k <= 46)
  );
}

function normalizeKey() {
  return true;
}

export function keyCode2String(keyCode, shift, spaceStatus) {
  if (keyCode > 0 && keyCode < 128) {
    return String.fromCharCode(keyCode);
  }

  switch (keyCode) {
    case Keys.FengHao:
      return shift ? ':' : ';';
    default:
      return '';
  }
}

export default Keys;
