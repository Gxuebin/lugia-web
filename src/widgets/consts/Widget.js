/**
 *
 * create by ligx
 *
 * @flow
 */
export const Input = 'sv.widget.Input';
export const Align = 'sv.widget.Align';
export const Trigger = 'sv.widget.Trigger';
export const InputTag = 'sv.widget.InputTag';
export const Theme = 'sv.widget.Theme';
export const Menu = 'sv.widget.Menu';
export const DropMenu = 'sv.widget.DropMenu';
export const Tree = 'sv.widget.Tree';
export const ThrottleScroller = 'sv.widget.ThrottleScroller';
