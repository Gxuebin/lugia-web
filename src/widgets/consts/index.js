/**
 *
 * create by ligx
 *
 * @flow
 */
const Button = 'sv_widget_Button';
const Input = 'sv_widget_Input';
const Align = 'sv_widget_Align';
const Trigger = 'sv_widget_Trigger';
const Tooltip = 'sv_widget_Tooltip';
const ThemeWrapWidget = 'sv_widget_ThemeWrapWidget';

const InputTag = 'sv_widget_InputTag';
const InputTagCloseButton = 'sv_widget_InputTagCloseButton';
const InputTagFontItem = 'sv_widget_InputTagFontItem';
const InputTagFocuInput = 'sv_widget_InputTagFocuInput';
const InputTagMoreItem = 'sv_widget_InputTagMoreItem';
const ItemTagOption = 'sv_widget_ItemTagOption';
const InputTagItem = 'sv_widget_InputTagItem';
const InputTagClearButton = 'sv_widget_InputTagClearButton';

const Theme = 'sv_widget_Theme';
const Menu = 'sv_widget_Menu';
const MenuItem = 'sv_widget_MenuItem';
const Icon = 'sv_widget_Icon';
const AddIcon = 'sv_widget_AddIcon';
const RefershIcon = 'sv_widget_RefershIcon';
const CheckIcon = 'sv_widget_CheckIcon';
const ClearIcon = 'sv_widget_ClearIcon';
const DropMenu = 'sv_widget_DropMenu';
const Tree = 'sv_widget_Tree';
const TreeSelect = 'sv_widget_TreeSelect';
const TreeSelectLimitTitle = 'sv_widget_TreeSelectLimitTitle';
const ThrottleScroller = 'sv_widget_ThrottleScroller';

/* add by wangcuixia */
const Switch = 'sv_widget_Switch';
const Slider = 'sv_widget_Slider';
const SliderButton = 'sv_widget_Slider_Button';
const SliderIcon = 'sv_widget_Slider_Icon';

/* add by liangguodong */
const SearchIcon = 'sv_widget_SearchIcon';
const PullIcon = 'sv_widget_PullIcon';

const ButtonGroup = 'sv_widget_ButtonGroup';
const Radio = 'sv_widget_Radio';
const RadioGroup = 'sv_widget_RadioGroup';
const CheckBox = 'sv_widget_CheckBox';
const CheckBoxGroup = 'sv_widget_CheckBoxGroup';
const CheckButton = 'sv_widget_CheckButton';
const AmountInput = 'sv_widget_AmountInput';
const Col = 'sv_widget_Col';
const Row = 'sv_widget_Row';
const NumberInput = 'sv_widget_NumberInput';
const Divider = 'sv_widget_Divider';
const Panel = 'sv_widget_Panel';
const Collapse = 'sv_widget_Collapse';
const Avatar = 'sv_widget_Avatar';
const Badge = 'sv_widget_Badge';
const NumberTurn = 'sv_widget_NumberTurn';

/* add by szfeng */
const Select = 'sv_widget_Select';
const SelectLimitTitle = 'sv_widget_SelectLimitTitle';
const QueryInput = 'sv_widget_QueryInput';
const Tag = 'sv_widget_Tag';
const DropMenuButton = 'DropMenuButton';

const Layout = 'sv_widget_Layout';
const Header = 'sv_widget_Header';
const Content = 'sv_widget_Content';
const Footer = 'sv_widget_Footer';
const Aside = 'sv_widget_Aside';
const Affix = 'sv_widget_Affix';

const Tabs = 'sv_widget_Tabs';
const Tabpano = 'sv_widget_Tabpano';
const TabsHContainer = 'sv_widget_TabsHContainer';

export default {
  Button,
  Input,
  Align,
  Trigger,
  Tooltip,
  ThemeWrapWidget,
  InputTag,
  InputTagCloseButton,
  InputTagFontItem,
  InputTagFocuInput,
  InputTagMoreItem,
  ItemTagOption,
  InputTagItem,
  InputTagClearButton,
  Theme,
  Menu,
  MenuItem,
  Icon,
  AddIcon,
  RefershIcon,
  CheckIcon,
  ClearIcon,
  DropMenu,
  Tree,
  TreeSelect,
  TreeSelectLimitTitle,
  ThrottleScroller,
  Switch,
  Slider,
  SliderButton,
  SliderIcon,
  SearchIcon,
  PullIcon,
  Radio,
  RadioGroup,
  ButtonGroup,
  CheckBox,
  CheckBoxGroup,
  CheckButton,
  AmountInput,
  Col,
  Row,
  NumberInput,
  Divider,
  Panel,
  Collapse,
  Avatar,
  Badge,
  NumberTurn,
  Select,
  SelectLimitTitle,
  Layout,
  Header,
  Content,
  Footer,
  Aside,
  QueryInput,
  Tabs,
  Tabpano,
  TabsHContainer,
  Affix,
  Tag,
  DropMenuButton,
};
