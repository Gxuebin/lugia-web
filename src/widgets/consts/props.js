/**
 *
 * create by ligx
 *
 * @flow
 */
export const DisplayField = 'text';
export const ValueField = 'value';
