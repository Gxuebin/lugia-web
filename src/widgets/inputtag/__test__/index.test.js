//@flow
import React from 'react';
import InputTag, { _InputTag_, } from '../';
import chai from 'chai';
import 'jest-styled-components';
import Enzyme, { mount, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Theme from '../../theme';
import { createTestComponent, delay, } from 'sv-test-utils';
import * as Widgets from '../../consts/Widget';

const { mockFunction, mockObject, VerifyOrder, VerifyOrderConfig, } = require('vx-mock');

Enzyme.configure({ adapter: new Adapter(), });

const { expect: exp, } = chai;

describe('InputTag', () => {
  const displayValue = '常,乐,我,净';
  const value = '1,2,3';

  class InputTagTest extends React.Component<any, any> {
    inputtag: any;

    render () {
      return <Theme config={{ [Widgets.InputTag]: { width: 300, }, }} {...this.props}>
        <InputTag value={value}
                  ref={cmp => this.inputtag = cmp}
                  displayValue={displayValue}/></Theme>;
    }
  }

  it('展现值&实际值 宽度够容纳所有的3个结点', async () => {
    const mockInputTagPrototye = mockObject.create(_InputTag_.prototype);
    const funcMock = mockInputTagPrototye.mockFunction('getFontWidth');
    funcMock.forever(5);


    const cmp = mount(<InputTagTest/>);
    await delay(100, () => {
      cmp.instance().forceUpdate();
      cmp.update();
      const tagItems = cmp.find(Widgets.InputTagItem);
      exp(tagItems.length).to.be.equal(4);
      exp(tagItems.at(1).text()).to.be.equal('常');
      exp(tagItems.at(2).text()).to.be.equal('乐');
      exp(tagItems.at(3).text()).to.be.equal('我');
      exp(cmp.find(Widgets.FontItem).at(0).getDOMNode() == tagItems.at(0).getDOMNode()).to.be.true;
      exp(cmp.find(Widgets.FontItem).length).to.be.equal(1);
    });
    mockInputTagPrototye.resetAll();
  });
  it('展现值&实际值 3个一个结点也容不下', async () => {
    const mockInputTagPrototye = mockObject.create(_InputTag_.prototype);
    const funcMock = mockInputTagPrototye.mockFunction('getFontWidth');
    funcMock.forever(1000);

    const cmp = mount(<InputTagTest/>);
    await delay(100, () => {
      cmp.instance().forceUpdate();
      cmp.update();
      const tagItems = cmp.find(Widgets.InputTagItem);
      exp(tagItems.length).to.be.equal(2);
      exp(cmp.find(Widgets.FontItem).length).to.be.equal(1);
      exp(cmp.find(Widgets.FontItem).at(0).getDOMNode() == tagItems.at(0).getDOMNode()).to.be.true;
      exp(cmp.find(Widgets.MoreInputTagItem).length).to.be.equal(1);
      exp(cmp.find(Widgets.MoreInputTagItem).at(0).getDOMNode() == tagItems.at(1).getDOMNode()).to.be.true;
    });
    mockInputTagPrototye.resetAll();
  });
  it('展现值&实际值 3个只能容纳一个', async () => {
    const mockInputTagPrototye = mockObject.create(_InputTag_.prototype);
    const funcMock = mockInputTagPrototye.mockFunction('getFontWidth');
    let i = 0;
    const val = [5, 1000, 1000,];
    funcMock.mock(() => {
      return val[ i++ ];
    });

    const cmp = mount(<InputTagTest/>);
    await delay(100, () => {
      cmp.instance().forceUpdate();
      cmp.update();
      const tagItems = cmp.find(Widgets.InputTagItem);
      exp(tagItems.length).to.be.equal(3);
      exp(cmp.find(Widgets.FontItem).length).to.be.equal(1);
      exp(cmp.find(Widgets.MoreInputTagItem).length).to.be.equal(1);
      exp(cmp.find(Widgets.FontItem).at(0).getDOMNode() == tagItems.at(0).getDOMNode()).to.be.true;
      exp(tagItems.at(1).text()).to.be.equal('常');
      exp(cmp.find(Widgets.MoreInputTagItem).at(0).getDOMNode() == tagItems.at(2).getDOMNode()).to.be.true;
    });
    mockInputTagPrototye.resetAll();
  });

});
