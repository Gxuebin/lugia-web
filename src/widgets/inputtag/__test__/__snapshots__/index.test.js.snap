// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTag InputTag items: 1 item,  can see  0 1`] = `
.c0 {
  width: 200px;
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  font-size: 12px;
}

.c1 {
  border: solid 1px #d9d9d9;
  border-radius: 4px;
  min-height: 28px;
  padding-bottom: 3px;
}

.c1:hover {
  border-color: #49a9ee;
}

.c2 {
  width: 212px;
  height: 26px;
  margin-left: 5px;
  margin-right: 7px;
  margin-bottom: -3px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c4 {
  margin-top: 3px;
  height: 20px;
  line-height: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #f3f3f3;
  border-radius: 4px;
  color: rgba(0,0,0,0.65);
  cursor: default;
  float: left;
  margin-right: 4px;
  position: relative;
  overflow: hidden;
  -webkit-transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  padding: 0 20px 0 10px;
}

.c5 {
  height: 20px;
  line-height: 20px;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  padding: 0;
}

.c6 {
  font-size: 14px;
  color: rgba(0,0,0,0.47);
  position: absolute;
  padding: 0 0 0 5px;
  right: 4px;
  zoom: 1;
}

.c6:hover {
  color: #000;
}

<span>
  <div
    className="sv c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <ul
          className="c3"
        >
          <li
            className="hidden c4"
            onClick={undefined}
          >
            <span
              className="c5"
            >
              123456789012345676890780dfasfasfa
            </span>
            <span
              className="iconfont icon-close c6"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</span>
`;

exports[`InputTag InputTag items: 1 item,  can see  1 1`] = `
.c0 {
  width: 200px;
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  font-size: 12px;
}

.c1 {
  border: solid 1px #d9d9d9;
  border-radius: 4px;
  min-height: 28px;
  padding-bottom: 3px;
}

.c1:hover {
  border-color: #49a9ee;
}

.c2 {
  width: 212px;
  height: 26px;
  margin-left: 5px;
  margin-right: 7px;
  margin-bottom: -3px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c4 {
  margin-top: 3px;
  height: 20px;
  line-height: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #f3f3f3;
  border-radius: 4px;
  color: rgba(0,0,0,0.65);
  cursor: default;
  float: left;
  margin-right: 4px;
  position: relative;
  overflow: hidden;
  -webkit-transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  padding: 0 20px 0 10px;
}

.c5 {
  height: 20px;
  line-height: 20px;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  padding: 0;
}

.c6 {
  font-size: 14px;
  color: rgba(0,0,0,0.47);
  position: absolute;
  padding: 0 0 0 5px;
  right: 4px;
  zoom: 1;
}

.c6:hover {
  color: #000;
}

<span>
  <div
    className="sv c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <ul
          className="c3"
        >
          <li
            className="hidden c4"
            onClick={undefined}
          >
            <span
              className="c5"
            >
              abcdeddfasdddfadasf
            </span>
            <span
              className="iconfont icon-close c6"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</span>
`;

exports[`InputTag InputTag items: 3 item,  can see  0 1`] = `
.c0 {
  width: 200px;
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  font-size: 12px;
}

.c1 {
  border: solid 1px #d9d9d9;
  border-radius: 4px;
  min-height: 28px;
  padding-bottom: 3px;
}

.c1:hover {
  border-color: #49a9ee;
}

.c2 {
  width: 212px;
  height: 26px;
  margin-left: 5px;
  margin-right: 7px;
  margin-bottom: -3px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c4 {
  margin-top: 3px;
  height: 20px;
  line-height: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #f3f3f3;
  border-radius: 4px;
  color: rgba(0,0,0,0.65);
  cursor: default;
  float: left;
  margin-right: 4px;
  position: relative;
  overflow: hidden;
  -webkit-transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  padding: 0 20px 0 10px;
}

.c5 {
  height: 20px;
  line-height: 20px;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  padding: 0;
}

.c6 {
  font-size: 14px;
  color: rgba(0,0,0,0.47);
  position: absolute;
  padding: 0 0 0 5px;
  right: 4px;
  zoom: 1;
}

.c6:hover {
  color: #000;
}

<span>
  <div
    className="sv c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <ul
          className="c3"
        >
          <li
            className="hidden c4"
            onClick={undefined}
          >
            <span
              className="c5"
            >
              123456789012345676890780
            </span>
            <span
              className="iconfont icon-close c6"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</span>
`;

exports[`InputTag InputTag items: 3 item,  can see  1 1`] = `
.c0 {
  width: 200px;
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  font-size: 12px;
}

.c1 {
  border: solid 1px #d9d9d9;
  border-radius: 4px;
  min-height: 28px;
  padding-bottom: 3px;
}

.c1:hover {
  border-color: #49a9ee;
}

.c2 {
  width: 212px;
  height: 26px;
  margin-left: 5px;
  margin-right: 7px;
  margin-bottom: -3px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c4 {
  margin-top: 3px;
  height: 20px;
  line-height: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #f3f3f3;
  border-radius: 4px;
  color: rgba(0,0,0,0.65);
  cursor: default;
  float: left;
  margin-right: 4px;
  position: relative;
  overflow: hidden;
  -webkit-transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  padding: 0 20px 0 10px;
}

.c5 {
  height: 20px;
  line-height: 20px;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  padding: 0;
}

.c6 {
  font-size: 14px;
  color: rgba(0,0,0,0.47);
  position: absolute;
  padding: 0 0 0 5px;
  right: 4px;
  zoom: 1;
}

.c6:hover {
  color: #000;
}

<span>
  <div
    className="sv c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <ul
          className="c3"
        >
          <li
            className="hidden c4"
            onClick={undefined}
          >
            <span
              className="c5"
            >
              12345678901234
            </span>
            <span
              className="iconfont icon-close c6"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</span>
`;

exports[`InputTag InputTag items: 3 item,  can see  1 more item 1`] = `
.c0 {
  width: 200px;
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  font-size: 12px;
}

.c1 {
  border: solid 1px #d9d9d9;
  border-radius: 4px;
  min-height: 28px;
  padding-bottom: 3px;
}

.c1:hover {
  border-color: #49a9ee;
}

.c2 {
  width: 212px;
  height: 26px;
  margin-left: 5px;
  margin-right: 7px;
  margin-bottom: -3px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c4 {
  margin-top: 3px;
  height: 20px;
  line-height: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #f3f3f3;
  border-radius: 4px;
  color: rgba(0,0,0,0.65);
  cursor: default;
  float: left;
  margin-right: 4px;
  position: relative;
  overflow: hidden;
  -webkit-transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  padding: 0 20px 0 10px;
}

.c5 {
  height: 20px;
  line-height: 20px;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  padding: 0;
}

.c6 {
  font-size: 14px;
  color: rgba(0,0,0,0.47);
  position: absolute;
  padding: 0 0 0 5px;
  right: 4px;
  zoom: 1;
}

.c6:hover {
  color: #000;
}

<span>
  <div
    className="sv c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <ul
          className="c3"
        >
          <li
            className="hidden c4"
            onClick={undefined}
          >
            <span
              className="c5"
            >
              abcdeddfasdddfadasf
            </span>
            <span
              className="iconfont icon-close c6"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</span>
`;

exports[`InputTag InputTag items: 3 item,  can see  3 1`] = `
.c0 {
  width: 200px;
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  font-size: 12px;
}

.c1 {
  border: solid 1px #d9d9d9;
  border-radius: 4px;
  min-height: 28px;
  padding-bottom: 3px;
}

.c1:hover {
  border-color: #49a9ee;
}

.c2 {
  width: 212px;
  height: 26px;
  margin-left: 5px;
  margin-right: 7px;
  margin-bottom: -3px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c4 {
  margin-top: 3px;
  height: 20px;
  line-height: 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #f3f3f3;
  border-radius: 4px;
  color: rgba(0,0,0,0.65);
  cursor: default;
  float: left;
  margin-right: 4px;
  position: relative;
  overflow: hidden;
  -webkit-transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: padding 0.3s cubic-bezier(0.645,0.045,0.355,1);
  padding: 0 20px 0 10px;
}

.c5 {
  height: 20px;
  line-height: 20px;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  padding: 0;
}

.c6 {
  font-size: 14px;
  color: rgba(0,0,0,0.47);
  position: absolute;
  padding: 0 0 0 5px;
  right: 4px;
  zoom: 1;
}

.c6:hover {
  color: #000;
}

<span>
  <div
    className="sv c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <ul
          className="c3"
        >
          <li
            className="hidden c4"
            onClick={undefined}
          >
            <span
              className="c5"
            >
              a
            </span>
            <span
              className="iconfont icon-close c6"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</span>
`;
